fbw-a380x/src/wasm/systems/a380_systems/src/landing_gear/mod.rs

```diff
@@ -0,0 +1,1683 @@
+            gear_system_control: GearSystemStateMachine::default(),
+            is_gear_lever_down: true,
+
+            transition_duration: Duration::default(),
+            gear_lever_position_is_down_previous_state: true,
+            status: LgciuStatus::Ok,
+
+            power_supply_failure: Failure::new(FailureType::LgciuPowerSupply(lgciu_id)),
+            internal_error_failure: Failure::new(FailureType::LgciuInternalError(lgciu_id)),
+
+            is_active_computer_previous_state: lgciu_id == LgciuId::Lgciu1,
+
+            should_open_doors: false,
+            should_extend_gears: context.start_gear_down(),
+        }
+    }
+
+    pub fn update(
+        &mut self,
+        context: &UpdateContext,
+        gear_system_sensors: &impl GearSystemSensors,
+        external_power_available: bool,
+        gear_handle: &impl LandingGearHandle,
+        is_master_computer: bool,
+    ) {
+        self.is_gear_lever_down = gear_handle.gear_handle_is_down();
+        self.external_power_available = external_power_available;
+
+        self.sensor_inputs.update(
+            gear_system_sensors,
+            external_power_available,
+            self.is_powered,
+        );
+
+        if self.is_powered {
+            if is_master_computer && !self.is_active_computer_previous_state {
+                self.actions_when_becoming_master();
+            }
+
+            if !is_master_computer {
+                self.actions_when_slave();
+            }
+
+            if !self.is_powered_previous_state {
+                self.actions_when_startup();
+            }
+
+            if is_master_computer {
+                self.monitor_gear_state();
+
+                // Only allow statemachine to work if status is OK. If not outputs will be frozen at old value
+                if self.status() == LgciuStatus::Ok {
+                    self.gear_system_control
+                        .update(&self.sensor_inputs, !self.is_gear_lever_down);
+
+                    self.compute_open_doors_output();
+                    self.compute_extend_gears_output();
+                }
+            }
+        }
+
+        if context.is_sim_ready() {
+            self.update_monitoring(context, gear_handle);
+        }
+
+        self.is_active_computer_previous_state = is_master_computer;
+        self.is_powered_previous_state = self.is_powered;
+    }
+
+    fn update_monitoring(&mut self, context: &UpdateContext, gear_handle: &impl LandingGearHandle) {
+        self.update_transition_timer(context, gear_handle);
+
+        if !self.is_powered {
+            self.status = LgciuStatus::FailedNotPowered;
+        } else if self.status == LgciuStatus::Ok {
+            if self.transition_duration > Self::MAX_TRANSITION_DURATION {
+                self.status = LgciuStatus::FailedNoChangeOver;
+            }
+
+            if self.internal_error_failure.is_active() {
+                self.status = LgciuStatus::FailedAutoChangeOver;
+            }
+        }
```

Comment:
```suggestion
       self.status = if !self.is_powered {
            LgciuStatus::FailedNotPowered
        } else {
            if self.internal_error_failure.is_active() {
                LgciuStatus::FailedAutoChangeOver
            } else if self.transition_duration > Self::MAX_TRANSITION_DURATION {
                LgciuStatus::FailedNoChangeOver
            } else {
                LgciuStatus::Ok
            }
        };
```

Reply:
Should be equivalent (use simple assignment)

Reply:
Shouldn't clippy do this? Also, you have an else with if -> else if -> else inside; that can still be simplified more :P.